libs/utils/src/failpoint_support.rs

```diff
@@ -33,6 +33,29 @@ macro_rules! __failpoint_sleep_millis_async {
 }
 pub use __failpoint_sleep_millis_async as sleep_millis_async;
 
+#[macro_export]
+macro_rules! __failpoint_sleep_millis_cancellable_async {
+    ($name:literal, $cancel:expr) => {{
```

Comment:
I wonder if this should just be another arm of the `__failpoint_sleep_millis_async` macro.

Also, on a more general note, the rooted paths starting with `::` help but the "proper" way to use stuff in macros hygenically is to use `$crate` together with a `#[doc(hidden)] mod _private` and then you do `$crate::_private::Option::Some` etc.

Reply:
(IMO the second point is orthogonal to what this PR is doing and also affects already existing codes so I guess it's not needed)

Reply:
> I wonder if this should just be another arm of the __failpoint_sleep_millis_async macro.

Good suggestion -- I rarely write macros, so forget that then can have variadic arguments.  1c259fd9311e58ec6c5cff735498aeb90d51a14e